alertwildfire_live_recording.py

```diff
@@ -127,40 +129,98 @@ def fetchAllCameras(camera_names_to_watch):
+        ["c", "cleaning_threshold", "time in hours to store data"],
         ["o", "cameras_overide", "specific cameras to watch"],
-        ["p", "parallelize", "toggle parallelisation"]
+        ["p", "parallelize", "toggle parallelisation"],
+        ["a", "agents", "number of agents to assign for parallelization"]
     ]
     args = collect_args.collectArgs(reqArgs,  optionalArgs=optArgs, parentParsers=[goog_helper.getParentParser()])
     
+    
+    if args.cleaning_threshold:
+        googleServices = goog_helper.getGoogleServices(settings, args)
+        cleaning_threshold = float(args.cleaning_threshold)
+        cleanup_archive(googleServices, cleaning_threshold)
     if args.cameras_overide:
         listofRotatingCameras = list(args.cameras_overide.replace(" ", "").strip('[]').split(','))
     else:
         listofCameras = alertwildfire_API.get_all_camera_info()
         listofRotatingCameras = [camera["name"] for camera in listofCameras if (camera["name"][-1]=='2') ]
+    if args.agents:
+        agents = int(args.agents)
+    else:
+        agents = None
     if args.parallelize:
-        parallel = args.parallelize
+        parallel = True
+        #num of camera's per process
+        test = "Axis-Briar2"
+        googleServices = goog_helper.getGoogleServices(settings, args)
+        dbManager = db_manager.DbManager(sqliteFile=settings.db_file,
+                                        psqlHost=settings.psqlHost,                  psqlDb=settings.psqlDb,    
+                                        psqlUser=settings.psqlUser, psqlPasswd=settings.psqlPasswd)
+        temporaryDir = tempfile.TemporaryDirectory()
+        tic = time.time()
+        capture_and_record(googleServices, dbManager, temporaryDir.name, test)
+        toc =time.time()-tic
+        # target estimate of camera refresh time
+        target_refresh_time_per_camera = 12#secs
+        num_cameras_per_process = math.floor(target_refresh_time_per_camera / toc)
     else:
         parallel = False
-
-    if parallel:#having issues
-        num_cameras_per_process = 5
-        camera_bunchs = [listofRotatingCameras[num_cameras_per_process*num:num_cameras_per_process*num+num_cameras_per_process] for num in range(0, math.ceil(len(listofRotatingCameras)/num_cameras_per_process))]
     
-        agents = 3
-        agents = len(camera_bunchs)
-        chunksize = 3
+
+    if parallel:
+        
+
+        #divy the cameras
+        camera_bunchs= []
+        num_of_processes_needed  =  math.ceil(len(listofRotatingCameras)/num_cameras_per_process)
+        if not agents:
+            agents = multiprocessing.cpu_count() -2#as not to inhibit computer functionality
+        if num_of_processes_needed>agents:
+            logging.warning('unable to process all cameras on this machine and maintain a target refresh rate of %s seconds, please reduce number of cameras to less than %s', target_refresh_time_per_camera,num_cameras_per_process*agents)
+            return
```

Comment:
OK, I see what you're doing now.  Personally, I would simplify everything and just require the agents parameter and base things off of that, and collect some stats on how fast we can process based on different agents values.